### DISCUSSION TITLE
Failed to Navigate Amazon Anti-Captcha Target

### DISCUSSION THREAD

**Discussion by ArmaanCPro** [jef/streetmerchant#2555] [Help]
So hi everyone! I'm sorry I'm asking so many questions, but how are we supposed to transverse the Amazon CAPTCHA? I posted another discussion about adding support for 2Captcha, you can see that here: https://github.com/jef/streetmerchant/discussions/2554. 
So if we weren't to use another CAPTCHA API, how are we supposed to solve Amazon CAPTCHAs? It is impossible in headful mode as Chromium just opens another tab while trying to do it. Even if you take the tab into another window and solve the CAPTCHA, which is still hard, Amazon will ask you for another CAPTCHA later on. Most of the time Amazon redirects you to dungeons and dragons or best deals or something. And the page shows the "Dogs of Amazon". 
Most of the time Amazon doesn't fail a CAPTCHA, but sometimes there is a little message in the terminal.
Thanks a lot :D

**Comment by klords** [comment]
> how are we supposed to solve Amazon CAPTCHAs? It is impossible in headful mode as Chromium just opens another tab while trying to do it.

https://www.jef.buzz/streetmerchant/reference/captcha/ we have an interactive captcha handler for exactly this scenario

The title of your post is for a different issue altogether. Long before the captcha handler was implemented, there was a "captcha-deterrent"/anti-captcha feature added to navigate to non-GPU related Amazon pages to try to reduce the number of captchas encountered. Sometimes it doesn't work or goes to an error page. It doesn't impact the bot's processing, so you can ignore these issues.

**Reply by ArmaanCPro** [reply]
Thank you, but I was asking a different question entirely. I know realize that I worded it way wrong, but I mean that how are we supposed to solve hCAPTCHA and reCAPTCHA?

**Reply by klords** [reply]
You asked multiple times specifically about Amazon. Amazon does not utilize hCaptcha or reCaptcha. Perhaps you're thinking of one of your other posts.

Regardless, there isn't a way to solve those captchas in SM yet. If you're facing them, you have to back off.
